execution/gethexec/blockchain.go

```diff
@@ -214,16 +222,35 @@ func GetBlockChain(chainDb ethdb.Database, cacheConfig *core.CacheConfig, chainC
 		EnablePreimageRecording: false,
 	}
 
+	if vmTraceConfig != nil && vmTraceConfig.TracerName != "" {
+		t, err := tracers.LiveDirectory.New(vmTraceConfig.TracerName, json.RawMessage(vmTraceConfig.JSONConfig))
+		if err == nil {
+			log.Info(vmTraceConfig.TracerName + " live tracer activated")
+			vmConfig.Tracer = t
+		} else {
+			log.Error("Custom tracer error: " + err.Error())
+		}
+	}
```

Comment:
As far as I can tell, this traces will never actually be used and you can avoid changing the blockchain. Not 100% certain, though.

Reply:
@tsahee I don't understand what you're saying here

I see that "live tracer activated" in the nitro logs when a tracer is compiled into nitro, this code does execute

Reply:
This code certainly gets called, but the blockchain itself doesn't use vmConfig.Tracer in any meaningful way in the codepath relevant to us.
It's not worth setting the tracer in blockchain if it's only use is to fetch it from the blockchain inside ProduceBlock.